### PR TITLE
Remove a redundant remove.

### PR DESCRIPTION
**Pull request by DanVanAtta**
[Review on Reviewable](https://reviewable.io/reviews/triplea-game/triplea/198)

### CODE REVIEW

File: src/games/strategy/net/ServerMessenger.java [hunk 1]
```diff
@@ -716,9 +716,6 @@ public void removeConnection(final INode node) {
     NotifyPlayerRemoval(node);
     SocketChannel channel = m_nodeToChannel.remove(node);
     if (channel == null) {
-      channel = m_nodeToChannel.remove(node);
```
Comment (DanVanAtta) [inline src/games/strategy/net/ServerMessenger.java]: This remove already happened on in line 717 two ines up

**Comment by ron-murhammer** [comment]
@DanVanAtta - Yeah I have no idea why it would try to remove it again if it returned null the first time.

@veqryn - I'll leave this for you to take a look at.

**Comment by veqryn** [comment]
@ron-murhammer, @DanVanAtta,
So it looks like the redundant remove was added in this commit in 2009 by @sbridges : 05acf38c6e4fb38b6e9cc99ea6062e9c44f4ad04
He specifically added both the first test for null, the second remove, then the second test for null and logging and returning.

It seems like this was done on purpose, not as a mistake.  This does not mean it isn't a mistake though, but I really wish I knew why this was done.  It certainly looks redundant.

Thoughts?

**Comment by DanVanAtta** [comment]
I think it was a mistake, it was perhaps initially intended to be the second parallel data structure:
```
    private ConcurrentHashMap<INode, SocketChannel> m_nodeToChannel = new ConcurrentHashMap<INode, SocketChannel>();
    private ConcurrentHashMap<SocketChannel, INode> m_channelToNode = new ConcurrentHashMap<SocketChannel, INode>();    
```

I'm noticing the names of those two are similar and often they are flipped and need to be managed together. I think the second one was supposed to be the other data structure perhaps - so a copy paste error.

Even with that, it does look like the two data structures are still handled correctly, and that this extra remove should have no effect. It is confusing since at first it looks like the second data member variable, and then one wonders why or if it was done intentionally.

**Comment by sbridges** [comment]
Probably a mistake

**Comment by veqryn** [comment]
@sbridges: thx for stopping by to clear that up :)

**Pull request merged**